Architecture/TodoApp/android/core/src/test/kotlin/com/novoda/todoapp/tasks/presenter/TasksPresenterTest.kt

```diff
@@ -62,16 +64,34 @@ class TasksPresenterTest {
     fun given_ThePresenterIsPresenting_on_EmissionOfNewTasks_it_ShouldPresentTheTasksToTheView() {
         givenThePresenterIsPresenting()
 
-        tasksSubject.onNext(simpleTasks())
+        tasksEventSubject.onNext(nonEmptyIdleEventWith(simpleTasks()))
 
         Mockito.verify(displayer).display(simpleTasks())
     }
 
     @Test
+    fun given_ThePresenterIsPresenting_on_EmissionOfSomeNewDeletedLocallyTasks_it_ShouldPresentNonDeletedTasksToTheView() {
+        givenThePresenterIsPresenting()
+
+        tasksEventSubject.onNext(nonEmptyIdleEventWith(someDeletedLocallyTasks()))
+
+        Mockito.verify(displayer).display(nonDeletedTasksOnly())
+    }
+
+    @Test
+    fun given_ThePresenterIsPresenting_on_EmissionOfAllDeletedLocallyTasks_it_ShouldNotPresentToTheView() {
+        givenThePresenterIsPresenting()
+
+        tasksEventSubject.onNext(nonEmptyIdleEventWith(allDeletedLocallyTasks()))
+
+        Mockito.verify(displayer, never()).display(any(Tasks::class.java))
+    }
+
+    @Test
     fun given_ThePresenterIsPresenting_on_EmissionOfALoadingEventWithNoData_it_ShouldPresentTheLoadingScreen() {
         givenThePresenterIsPresenting()
 
-        tasksEventSubject.onNext(Event.loading())
+        tasksEventSubject.onNext(nonEmptyLoadingEvent())
```

Comment:
Turns out `nonEmptyLoadingEvent` was badly named - that method returned an empty loading event! Fixed in https://github.com/novoda/spikes/pull/113/commits/0d9a3975bad3ed1a0d852e40e3934a699d6a2d53